### DISCUSSION TITLE
After rainbow is integrated, ios calls the signature method to jump to the appstore address of metamask by default

### DISCUSSION THREAD

**Discussion by mcwxxx** [rainbow-me/rainbowkit#1286] [Feedback]
Hello, we integrated rainbowkit into our web dapp and everything went smoothly. However, when the iphone called metamask signature, it would jump to metamask appstore by default. The official document could not help us solve this problem, please help us, thank you!

**Comment by mcwxxx** [comment]
@rainbow-me/rainbowkit ^0.12.9
"wagmi": "^0.12.12"

**Comment by DanielSinclair** [comment]
@mcwxxx This is unfortunately an issue with MetaMask and Universal Links in Mobile Safari. MetaMask has misconfigured their "app link" to redirect to the App Store, whereas wallets like Rainbow and Coinbase will allow a user to fallback to a webpage if a Universal Link redirect has failed. There are typically two causes:
1) The dApp must use Wagmi's [Prepare Hooks](https://wagmi.sh/react/prepare-hooks) to prepare a transaction before the user confirms it. Otherwise Mobile Safari will block the redirect as they think the website is attempting to click-jack the user.
2) These failures can also occur if the user has inadvertently activated a feature in Mobile Safari that remembers their redirect preference after opening a MetaMask link in a New Tab. For example, the user clicks on a MetaMask wallet icon on a dApp, hold presses, and taps Open in New Tab. This will tell Safari that you always want to open the link in Safari, rather than being linked directly into MetaMask. The only way to revert this is to attempt to open the MetaMask link again and choose "Open in MetaMask" instead. RainbowKit prevents users from going down this flow, but unfortunately other wallet connector libraries do not.
